### PR TITLE
Cluster provision refactor

### PR DESCRIPTION
**Pull request by chetnieter**
This refactor takes some redundant code from the HPCCloud taskflows and moves here to a new intermediate class, ClusterProvisioningTaskFlow. This PR depends on Kitware/HPCCloud#489

### CODE REVIEW

File: cumulus/taskflow/core/__init__.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+
+    return cluster
+
+def create_girder_client(girder_api_url, girder_token):
+    client = GirderClient(apiUrl=girder_api_url)
+    client.token = girder_token
+
+    return client
+
+def _get_image(logger, profile, image_spec):
```
Comment (chetnieter) [inline cumulus/taskflow/core/__init__.py]: This does not seemed to be used in either hpccloud or cumulus. Can it be removed?

Comment (cjh1) [inline cumulus/taskflow/core/__init__.py]: Which function? _get_image(...) ? Yes the looking up of the AMI is now done in the playbook.

File: cumulus/taskflow/core/__init__.py [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+        # Update the status to terminating
+        body = {
+            'status': JobState.TERMINATING
+        }
+        client.patch(job_url, data=json.dumps(body))
+
+        terminate_job(
+            cluster, job, log_write_url=None,
+            girder_token=task.taskflow.girder_token)
```
Comment (cjh1) [inline cumulus/taskflow/core/__init__.py]: Now that we have this code in cumulus. I don't think it belongs in taskflow/core/__init__.py Lets move it to taskflow/clusters/ or something like that?

Comment (chetnieter) [inline cumulus/taskflow/core/__init__.py]: Ok - will do.

File: cumulus/taskflow/core/__init__.py (hunk 1 above)
Comment (chetnieter) [inline cumulus/taskflow/core/__init__.py]: Yes _get_image(). So I can remove it?

Comment (cjh1) [inline cumulus/taskflow/core/__init__.py]: Yep

**Comment by chetnieter** [comment]
@cjh1 PTAL - I was able run and visualize a pyfr simulation with this branch.

**Comment by cjh1** [comment]
LGTM

**Pull request merged**